enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,306 @@
+---
+title: inject-cross-namespace-secrets-onto-openshift-hosts-for-general-pod-consumption
+
+authors:
+  - "@gabemontero"
+
+reviewers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@gnufied"
+  
+approvers:
+  - "@bparees"
+  - "@adambkaplan"
+  - "@derekwaynecarr"
+  - "@gnufied"
+  
+creation-date: 2020-03-17
+
+last-updated: 2020-03-18
+
+<!-- status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced -->
+status: provisional
+
+
+  
+---
+
+# Inject cross namespace secrets onto OpenShift Hosts for general Pod consumption
```

Comment:
Thanks for the suggestion @adambkaplan 

Though in cross referencing with other comments, like @bparees 's https://github.com/openshift/enhancements/pull/250#discussion_r394767547

`arguably this proposal(specific to the CSI driver) doesn't concern itself with whether the content is "global" or user namespaced.`

I'm inclined to *NOT* references namesapces in the title, and not use your suggestions as is.

Names and titles can be difficult, for sure.

My counter proposals:
-  `Expose Secrets/ConfigMap data via a CSI Driver to Pods`
-  `Share Secrets/ConfigMap data via a CSI Driver to Pods`
- `Expose Secrets/ConfigMaps via a CSI Driver to Pods`
- `Share Secrets/ConfigMaps via a CSI Driver to Pods`

Reply:
@gabemontero my point was that there's no actual concept of a "global" secret or a "namespaced" secret.  There are only secrets, they always live in namespaces, and if you want to mount one(that does not live in your pod's namespace) you will need this CSI driver.  Whether a secret is "global" or not is an artificial concept that will be determined by the admission control logic (e.g. treating all secrets that live in a particular special namespace as being "global" and thus always eligible for CSI driver copying, or treating all secrets annotated in a certain way as "global").



So referring to this CSI driver as allowing one to share a secret across namespaces is still reasonable, because even the "global" secrets are still, in reality, namespaced secrets that are being shared across namespaces (i.e. shared to pods that live in a different namespace).

Reply:
Ok thanks for the clarification @bparees 

So with that in mind, on possible titles, I currently see 
- yours in https://github.com/openshift/enhancements/pull/250#issuecomment-602845720 .. `CSI driver to enable pods to mount cross-namespace secrets`
- we've got @adambkaplan 's `Share Secrets And ConfigMaps Across Namespaces via a CSI Driver` earlier in this thread
- and we got my 4 here in this thread

Care to pick which one you want us to settle with ?

Reply:
@adambkaplan's works for me.

Reply:
part of next push